packages/apps-engine/src/server/AppManager.ts

```diff
@@ -492,6 +496,37 @@ export class AppManager {
         return true;
     }
 
+    public async migrate(id: string): Promise<boolean> {
+        const app = this.apps.get(id);
+
+        if (!app) {
+            throw new Error(`No App by the id "${id}" exists.`);
+        }
+
+        await app.call(AppMethod.ONUPDATE).catch((e) => console.warn('Error while migrating:', e));
+
+        await this.purgeAppConfig(app, { keepScheduledJobs: true });
+
+        const storageItem = await this.appMetadataStorage.retrieveOne(id);
+
+        app.getStorageItem().marketplaceInfo = storageItem.marketplaceInfo;
+        await app.validateLicense().catch();
+
+        storageItem.migrated = true;
+        storageItem.signature = await this.getSignatureManager().signApp(storageItem);
+        // This is async, but we don't care since it only updates in the database
+        // and it should not mutate any properties we care about
+        const stored = await this.appMetadataStorage.update(storageItem).catch();
```

Comment:
Calling catch and passing undefined still ends up triggering an unhandled rejection

Reply:
Oops, I believe there are a bunch more places to fix this, then :grimacing: :grimacing: 
We're doing the same in `enable`, `disable`, `initializeApp` and `enableApp` methods in this file -- and I followed this same wrong "pattern"